hicommonwealth/commonwealth
# Multichain erc20 vote weight + Native ETH voting

## Link to Issue
Closes: #9849 
Closes: #9927 

## Description of Changes
- Adds support for chain selection on ERC20 weighted topics
- Adds support for native ETH for ERC20 weighted topics

## Test Plan
- Ensure you have >= 1 CMN token on base sepolia
- Create ERC20 topic with token address `0x429ae85883f82203D736e8fc203A455990745ca1` (CMN token) with chain set to base sepolia and vote multiplier of 1
  - Create thread on topic and upvote it, ensure that upvote weight matches the number of tokens you own
- Create ERC20 topic using native token, with chain set to base sepolia (or any other chain you have native ETH on)
  - Create thread on topic and upvote it, ensure that upvote weight matches number of *full* tokens you own (not fractional)

## Deployment Plan
N/A

## Other Considerations
- Maybe native ETH should be a new topic type and not use ERC20 type?
